src/Model/Join.php

```diff
@@ -271,16 +268,25 @@ protected function init(): void
     protected function initJoinHooks(): void
     {
         $this->onHookToOwnerEntity(Model::HOOK_AFTER_LOAD, \Closure::fromCallable([$this, 'afterLoad']));
-        $this->onHookToOwnerEntity(Model::HOOK_AFTER_UNLOAD, \Closure::fromCallable([$this, 'afterUnload']));
+
+        $createHookFxWithCleanup = function (string $methodName): \Closure {
+            return function (Model $entity, &...$args) use ($methodName): void {
+                try {
+                    $this->{$methodName}($entity, ...$args);
+                } finally {
+                    $this->unsetSaveBuffer($entity);
```

Comment:
With non-reverse and reverse join, I do not think this is correct, as the non-reverse join will cleanup the save buffer in `Model::HOOK_BEFORE_INSERT` hook.